test/network/test_message_buffer.cpp

```diff
@@ -0,0 +1,25 @@
+#include <fc/network/message_buffer.hpp>
```

Comment:
Since you did this. Go ahead and move the tests from eos repo here. `./unittests/message_buffer_tests.cpp`. 

Then you can easily do a more complete threading test:
```
BOOST_AUTO_TEST_CASE(message_buffer_multithread)
{
   std::vector<std::thread> threads;
   for( int i = 0; i < 10; ++i ) {
      std::thread t( []() {
         for( int i = 0; i < 10; ++i ) {
            message_buff_growth();
         }
      } );
      threads.emplace_back( std::move( t ) );
   }
   for( auto& t : threads ) {
      t.join();
   }
}
```

Where `message_buff_growth()` is just the existing test moved to a function.

Reply:
That won't quite work, because Boost.Test isn't thread-safe.

Reply:
ah ok